src/PerformanceTests/Utils/TaskHelper.cs

```diff
@@ -0,0 +1,25 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+public class TaskHelper
+{
+    static readonly int cores = Environment.ProcessorCount;
+
+    public static async Task ParallelFor(int count, Func<Task> action)
+    {
+        // If we have a batch of 10,000 and have 10 cores we do 10 iterations of 1,000 message chunks.
```

Comment:
@danielmarbach Was it really your intention to have this behavior? Super large sets would basically have the same behavior.